Towny/src/main/java/com/palmergames/bukkit/towny/command/TownyAdminCommand.java

```diff
@@ -1726,6 +1781,38 @@ private void townyAdminNationTransfterTown(CommandSender sender, Nation nation,
 		TownyMessaging.sendMsg(sender, Translatable.of("msg_join_nation", town.getName()));
 	}
 
+	private void parseAdminNationSet(CommandSender sender, Nation nation, String[] split) throws TownyException {
+		if (split[0].equalsIgnoreCase("foundingdate")) {
+			checkPermOrThrow(sender, PermissionNodes.TOWNY_COMMAND_TOWNYADMIN_NATION_SETFOUNDINGDATE.getNode());
+
+			// Handle incorrect number of arguments
+			if (split.length == 1) {
+				throw new TownyException(Translatable.of("msg_err_invalid_input", "Use /townyadmin nation [nationname] set foundingdate [unix_timestamp]"));
+			}
+			
+			long timestamp;
+			try {
+				// They will input a UNIX timestamp in seconds (e.g. https://www.unixtimestamp.com/)
+				// We use a UNIX timestamp in milliseconds, hence * 1000.
+				timestamp = Long.parseLong(split[1]) * 1000;
+			} catch (NumberFormatException ife) {
+				throw new TownyException(Translatable.of("msg_error_must_be_int"));
+			}
+			if (timestamp <= 0) {
+				throw new TownyException(Translatable.of("msg_err_amount_must_be_greater_than_zero"));
+			}
+			if (timestamp > System.currentTimeMillis()) {
+				throw new TownyException(Translatable.of("msg_err_time_cannot_be_in_the_future"));
+			}
+			nation.setRegistered(timestamp);
+			nation.save();
+
+			TownyMessaging.sendMsg(sender, Translatable.of("msg_nation_registered_set", nation, TownyFormatter.registeredFormat.format(nation.getRegistered())));

```

Comment:
Like the town version this should be extracted to a method.

Reply:
Done.